pkg/controller/propagator/propagation.go

```diff
@@ -350,3 +387,72 @@ func (r *ReconcilePolicy) handleDecision(instance *policiesv1.Policy, decision a
 	}
 	return nil
 }
+
+// a helper to quickly check if there are any templates in any of the policy templates
+func policyHasTemplates(instance *policiesv1.Policy) bool {
+	for _, policyT := range instance.Spec.PolicyTemplates {
+		if templates.HasTemplate(policyT.ObjectDefinition.Raw, templateCfg.StartDelim) {
+			return true
+		}
+	}
+	return false
+}
+
+func processTemplates(replicatedPlc *policiesv1.Policy, decision appsv1.PlacementDecision, rootPlcNamespace string) error {
```

Comment:
Could you please add a comment documenting this function? Especially noting that the special annotation is removed.